File: string_array/H_Index.java
class Solution {
    public int hIndex(int[] citations) {
        int low = 0;
        int high = citations.length;
        while (low < high) {
            int mid = (low + high + 1) / 2;
            int count = 0;
            for (int i = 0; i < citations.length; ++i) {
                if (citations[i] >= mid) {
                    ++count;
                }
            }
            if (count >= mid) {
                low = mid;
            }
            else {
                high = mid - 1;
            }
        }
        return low;
    }
}

File: string_array/H_Index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        low = 0
        high = len(citations)
        while (low < high):
            mid = (low + high + 1) // 2
            count = 0
            for i in citations:
                if (i >= mid):
                    count += 1
            if (count >= mid):
                low = mid
            else:
                high = mid - 1
        return low
        